core/io.go

```diff
@@ -61,6 +61,12 @@ func (rp *RESPParser) DecodeOne() (interface{}, error) {
 			}
 			return nil, err
 		}
+
+		// Handle the case where no data is read but no error is returned
```

Comment:
Should we also handle the following error cases to make the connection handling robust?
For example:
Connection reset.
Connection closed locally.
Read Timeout.

Here are my thoughts, let me know what do you think:
list of scenarios to handle:

1. EOF (peer closed connection): Detected by comparing the error to io.EOF.
5. Zero-byte read: A read of 0 bytes without an error. In case of IO multiplexed scenario this means a closed connection.
2. Timeout: Check if net.Error and checking if it's a timeout.
3. Connection closed locally: Detect by checking for syscall.EBADF or os.ErrClosed.
4. Connection reset: Detect by checking for syscall.ECONNRESET.
6. Other errors: Any other errors are reported as general read errors.

Reply:
Thanks for the comprehensive list of error scenarios @soumya-codes, how about we create a separate issue for these and tackle them there? It seems that this PR handles the critical issue reported in #196 and we can merge it. Let me know your thoughts on this.

Reply:
Sure, makes sense. I will go ahead and create an issue for the same.